packages/core/src/services/essentials/gradient-utils.ts

```diff
@@ -0,0 +1,76 @@
+// Internal Imports
+import { Service } from "../service";
+import { DOMUtils } from "../";
+
+export class GradientUtils extends Service {
+	static appendLinearGradient(configs) {
+		const lg = configs.svg
+			.append("defs")
+			.append("linearGradient")
+			.attr("id", configs.id)
+			.attr("x1", configs.x1)
+			.attr("x2", configs.x2)
+			.attr("y1", configs.y1)
+			.attr("y2", configs.y2);
+		for (const stop of configs.stops) {
+			lg.append("stop")
+				.attr("offset", stop.offset)
+				.style("stop-color", stop.color)
+				.style("stop-opacity", stop.opacity);
+		}
+	}
+
+	static need3Stops(domain) {
+		if (domain[0] < 0 && domain[1] > 0) {
+			return true;
+		}
+	}
+
+	static getOffsetRatio(dataset) {
+		const minAndMax = [
+			Math.min.apply(Math, dataset.map(function(o) { return o.value; })),
+			Math.max.apply(Math, dataset.map(function(o) { return o.value; }))
```

Comment:
multiple issues in the above 2 lines:
- you're hard-coding `.value`, however that field is overridable using `mapsTo` in the options. We cannot do that since the code will break for users who use `mapsTo`
- running `Math.min` & `Math.max` on a large data array can be very performance heavy, is this information something that we can grab from the model, or any of the axis domains?